src/le.py

```diff
@@ -3325,6 +3415,107 @@ def cmd_follow(args):
     request_follow(filename, name, type_opt)
 
 
+def cmd_follow_multilog(args):
+    """
+    Follow the log(s) as defined in string passed to agent with
+    the '--multilog' parameter included - modification of cmd_follow
+    """
+    #### NOTE THAT BEHAVIOUR IS PROMPT USER TO CONTINUE THEN CALL request_follow
+    ####  TO SET UP A LOG WITH THE PATHANME PASSED TO THE AGENT
+
+    if len(args) == 0:
+        die("Error: Specify the file name of the log to follow.")
+    if len(args) > 1:
+        die("Error: Too many arguments.\n"
+            "A common mistake is to use wildcards in path that is being "
+            "expanded by shell. Enclose the path in single quotes to avoid "
+            "expansion.")
+
+    config.load()
+    config.agent_key_required()
+    # FIXME: follow to add logs into local configuration
+
+    arg = args[0]
+    pname = os.path.abspath(arg)
+    pnames = glob.glob(pname)
+    if len(pnames) == 0:
+        die('\nWarning: No Files found for %s\n' % pname)
+    files_with_log_names = get_log_names_for_files(pnames)
+    # user input required before new files are set to be followed
+    files_to_follow = user_prompt(files_with_log_names)
+    debug
+    for file in files_to_follow:
+        print "request to follow %s " % file
+    ### AT THIS TIME IGNORING THE LIST OF FILES RETURNED
+    ### file_to_follow AS NEED TO CONFIRM AGENT BEHAVIOUR
+    name = config.name
+    if name == NOT_SET:
+        name = os.path.basename(pname)
+    type_opt = config.type_opt
+    if type_opt == NOT_SET:
+        type_opt = ""
+    filename = "Multiple:" + pname
+    request_follow(filename, name, type_opt)
+
+def user_prompt(files_and_log_names):
+    """
+    Displays 2 lists - files already followed with log names, and those not.
+    Prompts user if they wish to follow files or not
+    """
+    ### NOTE THAT THIS WILL NEED REVIEW AND EXPECT TO CHANGE
+    ### THE OUTPUT WAS MADE UP ON THE SPOT AND THE EXPECTED
+    ### OPTIONS MAY CHANGE AS WELL AS THE TEXT
+    ### THIS USER PROMPT MAY OVER COMPLICATED THINGS
+
+    files_already_followed = {}
+    files_not_followed = {}
+    for filename, logname in files_and_log_names.items():
```

Comment:
Better to use the python list comprehensions here
files_not_followed = {fname: logname for (fname, logname) in files_and_log_names.iteritems() if logname is NOT_SET}
files_followed = {fname: logname for (fname, logname) in files_and_log_names.iteritems() if fname not in files_not_followed}